sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_base.py

```diff
@@ -0,0 +1,289 @@
+        """
+        Creates the error retry policy.
+        :param ~azure.eventhub._configuration.Configuration config: Configuration.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def create_link_properties(link_properties):
+        """
+        Creates and returns the link properties.
+        :param dict[bytes, int] link_properties: The dict of symbols and corresponding values.
+        :rtype: dict
+        """
+
+    @staticmethod
+    @abstractmethod
+    def create_connection(**kwargs):
+        """
+        Creates and returns the uamqp Connection object.
+        :keyword str host: The hostname, used by uamqp.
+        :keyword JWTTokenAuth auth: The auth, used by uamqp.
+        :keyword str endpoint: The endpoint, used by pyamqp.
+        :keyword str container_id: Required.
+        :keyword int max_frame_size: Required.
+        :keyword int channel_max: Required.
+        :keyword int idle_timeout: Required.
+        :keyword Dict properties: Required.
+        :keyword int remote_idle_timeout_empty_frame_send_ratio: Required.
+        :keyword error_policy: Required.
+        :keyword bool debug: Required.
+        :keyword str encoding: Required.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def close_connection(connection):
+        """
+        Closes existing connection.
+        :param connection: uamqp or pyamqp Connection.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_connection_state(connection):
+        """
+        Gets connection state.
+        :param connection: uamqp or pyamqp Connection.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def create_send_client(*, config, **kwargs):
+        """
+        Creates and returns the send client.
+        :param ~azure.eventhub._configuration.Configuration config: The configuration.
+
+        :keyword str target: Required. The target.
+        :keyword JWTTokenAuth auth: Required.
+        :keyword int idle_timeout: Required.
+        :keyword network_trace: Required.
+        :keyword retry_policy: Required.
+        :keyword keep_alive_interval: Required.
+        :keyword str client_name: Required.
+        :keyword dict link_properties: Required.
+        :keyword properties: Required.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def send_messages(producer, timeout_time, last_exception, logger):
+        """
+        Handles sending of event data messages.
+        :param ~azure.eventhub._producer.EventHubProducer producer: The producer with handler to send messages.
+        :param int timeout_time: Timeout time.
+        :param last_exception: Exception to raise if message timed out. Only used by uamqp transport.
+        :param logger: Logger.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def set_message_partition_key(message, partition_key, **kwargs):
+        """Set the partition key as an annotation on a uamqp message.
+
+        :param message: The message to update.
```

Comment:
type here?

Reply:
Oh are we not adding them b/c there is a diff between uamqp and pyamqp?

Reply:
Yeah it can be either uamqp or pyamqp Messages. So I didn't add the type. I did add the type in their respective transports though.